boot/bootutil/src/bootutil_misc.c

```diff
@@ -387,3 +387,31 @@ boot_write_enc_key(const struct flash_area *fap, uint8_t slot,
     return 0;
 }
 #endif
+
```

Comment:
It'd be nice to make sure we've got a good match between this code and what is in https://github.com/mcu-tools/mcuboot/blob/main/sim/src/image.rs#L1516-L1534. Whether they should even be separate is another consideration.

I did do a little testing where I added '4' to the max size to make sure it was causing failures.